.orchestra/config/components/toolchain/msvc.yml

```diff
@@ -386,16 +386,23 @@ toolchain/(@= name @)/vc:
         }
         EOF
 
+          WINE_TMP_DIRECTORY_32="$(mktemp --tmpdir tmp.wine-prefix.XXXXXXXXXX -d)"
+          trap 'rm -rf "$WINE_TMP_DIRECTORY_32"' EXIT
+
           (@ if debug_info: @)
-          ./bin/(@= triple @)-cl test.c /DEBUG:FASTLINK /Zi
+          WINEPREFIX="$WINE_TMP_DIRECTORY_32" ./bin/(@= triple @)-cl test.c -nologo /DEBUG:FASTLINK /Zi
           (@ else: @)
-          ./bin/(@= triple @)-cl test.c
+          WINEPREFIX="$WINE_TMP_DIRECTORY_32" ./bin/(@= triple @)-cl test.c -nologo 
```

Comment:
Why are you setting it here? The `WINEPREFIX` we set in `components.yml` should be good for these.
I suggest not to touch `WINEPREFIX` here and just set it to a temporary directory below, before `if [[ "(@= triple @)" =~ ^x86_64- ]]; then`, without distinguishing between 32 and 64.

Reply:
I don't have any strong opinions on the `WINEPREFIX`, the only thing I know is that my local directory it points to got corrupted somehow. So I just "threw random stuff at it" until it worked.

Reply:
Ah, wait, I remember now! Since there are now multiple toolchain dependencies, doing `orc install 
 revng-qa` tries to do all of them in parallel. Which leads to trouble unless we use different directories.

Reply:
`pool: wine` should fix that.

Reply:
Wait, it can be used on orchestra configs too? Is there an example anywhere?

Reply:
Ah sorry, I mixed the PRs... I'm starting to think we need a `$WINPREFIX/.cl.exe.lock` containing the PID of the process holding the lock. If there's no such process with that PID, we can drop it. That would enable us to drop the `pool: wine` thing too.

But what you did works too, up to you.

Reply:
While on topic, we might really want to think about enabling multiple instances to run in parallel too. Because the `test/revng-qa` performance degraded by A LOT because of this.

If it's not super tricky (e.g. if we can get away with just multiple `$WINPREFIX` directories, it's definitely worth trying to do it), otherwise, we can maybe at least allow `pool: wine` tests to start running right away, instead of waiting for literally every other test to finish and then create bottleneck at the end.

Reply:
In my understanding, having a clean `$WINEPREFIX` for each invocation is slow. 

```
export WINPREFIX=/tmp/lol
time wine cmd.exe /c exit # 1st invocation, takes 1.27s
time wine cmd.exe /c exit # 2nd invocation, takes 0.26s
```

However, this is still probably significantly better than waiting as we're doing now.
It will also spawn *a lot* of `wine` support processes (`wineserver`). We should run `wineserver -k` (which shuts them down), if we want to this route.

I'm not opposed to this, we can go this route.

Reply:
> In my understanding, having a clean `$WINEPREFIX` for each invocation is slow.

Yep, it's about as slow as what we're currently doing (no measurements, but it didn't `feel` better).

I'd guess the "best" way would be to spawn as many of these as the number of threads we run in parallel and then let those get reused. But that doesn't sound trivial to manage. Not sure if ninja provides some nice facilities for that though - that would be really helpful.

Reply:
That is to say - this here is fine - it's only during toolchain installation - which we don't do that often, so being somewhat slow doesn't matter.

I'm talking about `pool:wine`'d tests.

Reply:
I looked around, and it looks like there's something else going on too. There are non-deterministic problems when multiple of these are ran in parallel using different `WINEPREFIX`es. I guess this will have to do for now.